src/pool/srv_target.c

```diff
@@ -1861,9 +1861,20 @@ cont_discard_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	param.ip_epr.epr_lo = 0;
 	param.ip_epr.epr_hi = arg->tgt_discard->epoch;
 	uuid_copy(arg->cont_uuid, entry->ie_couuid);
+	do {
+		/* Inform the iterator and delete the object */
+		*acts |= VOS_ITER_CB_DELETE;
+		rc = vos_iterate(&param, VOS_ITER_OBJ, false, &anchor, obj_discard_cb, NULL,
+				 arg, NULL);
+		if (rc != -DER_BUSY && rc != -DER_INPROGRESS)
+			break;
 
-	rc = vos_iterate(&param, VOS_ITER_OBJ, false, &anchor, obj_discard_cb, NULL,
-			 arg, NULL);
+		D_DEBUG(DB_REBUILD, "retry by "DF_RC"/"DF_UUID"\n",
+			DP_RC(rc), DP_UUID(entry->ie_couuid));
+		/* Busy - inform iterator and yield */
+		*acts |= VOS_ITER_CB_YIELD;
```

Comment:
You do not need to do this and it's less efficient to do so.   This flag doesn't tell the iterator to yield but tells the iterator that a yield actually happened and that a reprobe is required.

The iterator internally checks to see if the schedule count changed and sets this flag internally for all callbacks.   Thus if we yield but nothing else gets scheduled before this ULT comes back, we won't set the flag and avoid the re-probe.  We should probably clean these up everywhere.